agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -77,6 +77,9 @@ type CredentialSpecResource struct {
 	// * key := credentialspec:s3ARN, value := credentialspec=file://CredentialSpecResourceLocation/s3_taskARN_fileName.json
 	// * key := credentialspec:ssmARN, value := credentialspec=file://CredentialSpecResourceLocation/ssm_taskARN_param.json
 	CredSpecMap map[string]string
+	// datalock should protect the CredSpecMap
+	datalock sync.RWMutex
```

Comment:
could you also add comments on why this separate lock is needed?

Reply:
I added information to the commit message. 
Do you want it replicated here?

Reply:
that explains 'what' the locks are for. I'm looking for the answer for 'why'.

Reply:
Added more details.